Snyk workflow failing when running the checks against the Operator

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

ci

### Describe the bug

The Snyk analysis when running to check the Operator is failing since last month. It seems that the build command is unable to retrieve additional dependencies:

```
Failed to get dependencies for all 1 potential projects.
Tip: Re-run in debug mode to see more information: DEBUG=*snyk* <COMMAND>
If the issue persists contact support@snyk.io
```

Later on, the workflow fails because the SARIF file was not generated after the build:

```
Run github/codeql-action/upload-sarif@v2.1.37
Error: Path does not exist: operator-report.sarif
Error: Path does not exist: operator-report.sarif
    at getSarifFilePaths (/home/runner/work/_actions/github/codeql-action/v2.1.37/lib/upload-lib.js:137:15)
    at Object.uploadFromActions (/home/runner/work/_actions/github/codeql-action/v2.1.37/lib/upload-lib.js:132:30)
    at run (/home/runner/work/_actions/github/codeql-action/v2.1.37/lib/upload-sarif-action.js:46:47)
    at async runWrapper (/home/runner/work/_actions/github/codeql-action/v2.1.37/lib/upload-sarif-action.js:68:9)
```

### Version

20.0.3

### Expected behavior

- Snyk workflow to pass

### Actual behavior

- Snyk workflow failing

### How to Reproduce?

- See https://github.com/keycloak/keycloak/actions/workflows/snyk-analysis.yml

### Anything else?

_No response_

@stianst unfortunately, Snyk is failing to fetch some dependencies required for the Operator. Please see:

```
 snyk-test [INFO] ------------------------------------------------------------------------
  snyk-test [INFO] BUILD FAILURE
  snyk-test [INFO] ------------------------------------------------------------------------
  snyk-test [INFO] Total time:  07:56 min
  snyk-test [INFO] Finished at: 2023-01-25T19:09:53-03:00
  snyk-test [INFO] ------------------------------------------------------------------------
  snyk-test [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:tree (default-cli) on project keycloak-operator: Cannot build project dependency graph: Could not resolve nor collect following dependencies: [org.keycloak:keycloak-common:jar:999-SNAPSHOT (compile)]: Could not resolve dependencies for project org.keycloak:keycloak-operator:jar:999-SNAPSHOT: Could not find artifact org.keycloak:keycloak-common:jar:999-SNAPSHOT -> [Help 1]
  snyk-test [ERROR]
```

The quick fix is to bring back the build of the Operator:

```
mvn -Poperator -pl operator -am -DskipTests clean install
```

I will try to investigate if we have more alternatives.

--- a/.github/workflows/snyk-analysis.yml
+++ b/.github/workflows/snyk-analysis.yml
@@ -34,7 +34,9 @@ jobs:
           sarif_file: quarkus-report.sarif
 
       - name: Check for vulnerabilities in Operator
-        run: snyk test --policy-path=${GITHUB_WORKSPACE}/.github/snyk/.snyk --all-projects --prune-repeated-subdependencies --exclude=tests --sarif-file-output=operator-report.sarif operator
+        run: |
+          mvn -Poperator -pl operator -am -DskipTests clean install
+          snyk test --policy-path=${GITHUB_WORKSPACE}/.github/snyk/.snyk --all-projects --prune-repeated-subdependencies --exclude=tests --sarif-file-output=operator-report.sarif operator
         continue-on-error: true
         env:
           SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}

